dashboard/modules/job/job_manager.py

```diff
@@ -73,11 +76,17 @@ def get_last_n_log_lines(
     ) -> str:
         log_tail_iter = self.tail_logs(job_id)
         log_tail_deque = deque(maxlen=num_log_lines)
+        line_length = 0
         for line in log_tail_iter:
-            if line is None:
+            if line is None or line_length >= self.MAX_LOG_SIZE:
                 break
             else:
+                new_line_length = len(line) + line_length
+                if new_line_length > self.MAX_LOG_SIZE:
+                    line = line[0 : self.MAX_LOG_SIZE - line_length]
+                    new_line_length = len(line) + line_length
```

Comment:
Can you add a comment explaining what this is doing? It was a little hard to understand. Could you also add a docstring stating in the overflow case whether we're returning the first 20000 characters of the last n log lines, or just the last 20000 characters of the logs